spd/experiments/lm/ss_llama_single_with_comparison_config.yaml

```diff
@@ -0,0 +1,124 @@
+# --- WandB ---
+wandb_project: spd
+wandb_run_name: null
+wandb_run_name_prefix: ""
+
+# --- General ---
+seed: 0
+C: 4000
+n_mask_samples: 1
+gate_type: "vector_mlp"
+gate_hidden_dims: [12]
+sigmoid_type: "leaky_hard"
+target_module_patterns: ["model.layers.*.mlp.gate_proj", "model.layers.*.mlp.down_proj", "model.layers.*.mlp.up_proj", "model.layers.*.self_attn.q_proj", "model.layers.*.self_attn.k_proj", "model.layers.*.self_attn.v_proj", "model.layers.*.self_attn.o_proj"]
+sampling: "binomial"
+
+# --- Loss Coefficients ---
+faithfulness_coeff: 10000000.0
+recon_coeff: null
+stochastic_recon_coeff: 1.0
+recon_layerwise_coeff: null
+stochastic_recon_layerwise_coeff: 1.0
+importance_minimality_coeff: 0.0003
+schatten_coeff: null
+out_recon_coeff: null
+embedding_recon_coeff: null
+is_embed_unembed_recon: false
+pnorm: 2.0
+p_anneal_start_frac: 0.0
+p_anneal_final_p: 0.1
+p_anneal_end_frac: 1.0
+output_loss_type: kl
+
+# --- Training ---
+batch_size: 12
+eval_batch_size: 12
+steps: 300000
+lr: 0.0005
+lr_schedule: cosine
+lr_warmup_pct: 0.0
+lr_exponential_halflife: null
+gradient_accumulation_steps: 4
+
+# --- Logging & Saving ---
+train_log_freq: 100
+eval_freq: 100
+slow_eval_freq: 100
+slow_eval_on_first_step: true
+n_eval_steps: 5
+save_freq: null
+ci_alive_threshold: 0.0
+n_examples_until_dead: 1368400
+eval_metrics:
+  - classname: "CIHistograms"
+    extra_init_kwargs:
+      n_batches_accum: 5
+  - classname: "ComponentActivationDensity"
+    extra_init_kwargs: {}
+  - classname: "CI_L0"
+    extra_init_kwargs:
+      groups:
+        total: ["*"]  # Sum of all L0 values
+        layer_0: ["model.layers.0.*"]
+        layer_1: ["model.layers.1.*"]
+        layer_2: ["model.layers.2.*"]
+        layer_3: ["model.layers.3.*"]
+  - classname: "CEandKLLosses"
+    extra_init_kwargs:
+      rounding_threshold: 0.0
+  - classname: "SubsetReconstructionLoss"
+    extra_init_kwargs:
+      n_mask_samples: 1
+      use_all_ones_for_non_replaced: false
+      include_patterns:
+        layer_0_only: ["model.layers.0.*"]
+        layer_1_only: ["model.layers.1.*"]
+        layer_2_only: ["model.layers.2.*"]
+        layer_3_only: ["model.layers.3.*"]
+        mlp_only: ["*.mlp.*"]
+        attention_only: ["*.self_attn.*"]
+      exclude_patterns:
+        all_but_layer_0: ["model.layers.0.*"]
+        all_but_layer_1: ["model.layers.1.*"]
+        all_but_layer_2: ["model.layers.2.*"]
+        all_but_layer_3: ["model.layers.3.*"]
+  - classname: "GeometricSimilarityComparison"
+    extra_init_kwargs:
+      reference_run_path: "wandb:goodfire/spd/runs/2js1ccon"
```

Comment:
immediate thoughts: it strikes me as off having a reference run path in the config for a new run. The standard use case in my head after reading the PR description is to having a post-hoc script which does the comparisons when the runs have finished (or on intermediate checkpoints). The issue there is waiting for runs to finish (or wasting space with saving intermediate checkpoints).

Will think more when reviewing and afterwards. Interested in thoughts.